### PR TITLE
Enhancement - Simplify Setting of RequestContext Proxy

### PR DESCRIPTION
**Pull request by amaitland**
**Fixes:** 

Issue #3235 

**Summary:**
Simplifies the setting of proxy preference

**Changes:** 
- Add new `RequestContextHandler` default implementation
- Add new extension methods
- Add Fluent API for setting the proxy server

Example of new API looks like this:
```c#
var requestContext = RequestContext
	.Configure()
	.WithProxyServer("127.0.0.1", 8080)
	.Create();

using (var browser = new ChromiumWebBrowser("http://cefsharp.github.io/", requestContext: requestContext))
{
	await browser.LoadPageAsync();

	var mainFrame = browser.GetMainFrame();
	Assert.True(mainFrame.IsValid);
	Assert.Contains("cefsharp.github.io", mainFrame.Url);

	output.WriteLine("Url {0}", mainFrame.Url);
}
```
      
**How Has This Been Tested?**  
Basic Unit test (only runs locally) spawns a proxy server and makes a simple page load request.

**Types of changes**

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Updated documentation

**Checklist:**

- [ ] Tested the code(if applicable)
- [x] Commented my code
- [ ] Changed the documentation(if applicable)
- [x] New files have a license disclaimer
- [x] The formatting is consistent with the project (project supports .editorconfig)

### CODE REVIEW

File: CefSharp.Test/OffScreen/OffScreenBrowserBasicFacts.cs [hunk 1]
```diff
@@ -284,5 +285,24 @@ public async Task CanExecuteJavascriptInMainFrameAfterNavigatingToDifferentOrigi
                 Assert.True(browser.CanExecuteJavascriptInMainFrame);
             }
         }
+
+        [Fact]
+        public async Task CanLoadGoogleUsingProxy()
+        {
+            var requestContextHander = new RequestContextHandler()
```
Comment (campersau) [inline CefSharp.Test/OffScreen/OffScreenBrowserBasicFacts.cs:292]: `requestContextHandler`

**Comment by campersau** [comment]
[This test](https://github.com/cefsharp/CefSharp/pull/3250/files#diff-2a9dc3cc99be62a9cf57e66c274fff07R290) currently fails because it also requires a proxy (also the name is a bit misleading as it does not load google but cefsharp website).

The proxy tests can also run in CI as demonstrated here https://ci.appveyor.com/project/campersau/cefsharp/builds/35589085 . Revert the [last commit](https://github.com/cefsharp/CefSharp/pull/3250/commits/c0b22e20368f3f3cff40da673870a7ff713a4fda) but keep this line https://github.com/cefsharp/CefSharp/pull/3250/files#diff-a1bde5e93ad7852b38eee3a892bbcd36R43.

**Comment by amaitland** [comment]
> also the name is a bit misleading as it does not load google but cefsharp website).

Thanks, should be renamed. `google.com.au` requires `TLS` which requires the proxy to generate a certificate and asks to install it, so best avoided.

> The proxy tests can also run in CI as demonstrated here https://ci.appveyor.com/project/campersau/cefsharp/builds/35589085 . Revert the [last commit](https://github.com/cefsharp/CefSharp/pull/3250/commits/c0b22e20368f3f3cff40da673870a7ff713a4fda) but keep this line https://github.com/cefsharp/CefSharp/pull/3250/files#diff-a1bde5e93ad7852b38eee3a892bbcd36R43.

Thanks, I don't want to actually run a proxy as part of the CI.

I'm aware of the test failing and will fix it later, thanks.

**Comment by amaitland** [comment]
Converting to draft for now as this isn't quite ready for review yet.

@campersau Sorry if I came across as dismissive, review is most welcomed. Just not quite ready for that yet 😄

**Comment by amaitland** [comment]
Probably could use a few more unit tests, the basics are done. I plan to merge in the next few days.

Thoughts/comments welcome.

**Pull request merged**